modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/PublicationChannelToWorkspace.java

```diff
@@ -0,0 +1,153 @@
+import org.opencastproject.mediapackage.MediaPackageElementFlavor;
+import org.opencastproject.mediapackage.Publication;
+import org.opencastproject.workflow.api.AbstractWorkflowOperationHandler;
+import org.opencastproject.workflow.api.ConfiguredTagsAndFlavors;
+import org.opencastproject.workflow.api.WorkflowInstance;
+import org.opencastproject.workflow.api.WorkflowOperationException;
+import org.opencastproject.workflow.api.WorkflowOperationHandler;
+import org.opencastproject.workflow.api.WorkflowOperationInstance;
+import org.opencastproject.workflow.api.WorkflowOperationResult;
+import org.opencastproject.workflow.api.WorkflowOperationResult.Action;
+
+import org.apache.commons.lang3.StringUtils;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Workflow operation handler to move elements from publication channel to workspace
+ */
+@Component(
+    immediate = true,
+    service = WorkflowOperationHandler.class,
+    property = {
+        "service.description=move publication elements to workspace",
+        "workflow.operation=publication-channel-to-workspace"
+    }
+)
+
+public class PublicationChannelToWorkspace extends AbstractWorkflowOperationHandler {
+
+  /** Configuration key for the "source-channel" to use as a source input */
+  static final String OPT_SOURCE_PUBLICATION_CHANNEL = "source-channel";
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory
+          .getLogger(PublicationChannelToWorkspace.class);
+
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext context)
+      throws WorkflowOperationException {
+
+    logger.info("Running get Publicationchannel to workspace for medipackage {}", workflowInstance.getId());
+    final MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+
+    WorkflowOperationInstance currentOperation = workflowInstance.getCurrentOperation();
+
+    ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(workflowInstance, Configuration.many,
+        Configuration.many, Configuration.many, Configuration.many);
+    List<MediaPackageElementFlavor> configuredSourceFlavors = tagsAndFlavors.getSrcFlavors();
+    List<String> configuredSourceTags = tagsAndFlavors.getSrcTags();
+    List <String> configuredTargetTags = tagsAndFlavors.getTargetTags();
+    String publicationChannel = StringUtils
+        .trimToEmpty(currentOperation.getConfiguration(OPT_SOURCE_PUBLICATION_CHANNEL));
+    String configuredTargetTagsAsString =  String.join(",", configuredTargetTags);
+
+    if (publicationChannel.isEmpty()) {
+      logger.error("No source publication-channel set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
```

Comment:
This is a clear misconfiguration, and having the operation without this option set is clearly not what is intended. Therefore, I'm not a fan of this and would let the operation fail instead of skipping it.